### DISCUSSION TITLE
Manual dark mode

### DISCUSSION THREAD

**Discussion by motaylormo** [easepick/easepick#225] [General]
The [current rendition of dark mode](https://easepick.com/packages/amp-plugin.html#option-darkMode) is only dark "if the user's system settings are set to dark mode."

I'm trying to use easepick on a page which can be either `<html data-theme="light"> ... </html>` or `<html data-theme="dark"> ... </html>`. The user's system settings determines which mode the page is _initially_, but there's also a toggle button in the corner that allows users to change between the two manually. If the user sets _this page_ to dark — but doesn't make dark their preferred system setting overall — how do I make easepick dark?

**Comment by d2roth** [comment]
I haven't tested it but this looks like it should do what you want:  https://github.com/easepick/easepick/discussions/77#discussioncomment-2924747

They disabled the built-in detection and set the theme in the setup method. 

Here is the code from the demo:
```
 const picker = new easepick.create({
 	element: document.getElementById('datepicker'),
 	css: [
 		'https://cdn.jsdelivr.net/npm/@easepick/bundle@1.2.0/dist/index.css',
  ],
  plugins: ['AmpPlugin'],
  AmpPlugin: {
  	darkMode: false, // disable matchMedia listener
  },
  setup(picker) {
  	picker.ui.container.dataset.theme = 'dark'; // <-- This is the line you need to call when changing the site's theme
  }
});
```

I haven't personally used it though.

**Comment by cgsmith** [comment]
I solved it this way. I put it right after the `picker.on` call. In my updateTheme() function (where I set localStorage.theme) I also had to call `window.dispatchEvent( new Event('storage') );`

```js
  document.addEventListener('DOMContentLoaded', () => {
      window.addEventListener('storage', function () {
          picker.ui.container.dataset.theme = document.documentElement.getAttribute('data-theme');
      });
  });
```
